pkg/cluster/spec/spec.go

```diff
@@ -129,6 +129,8 @@ type Topology interface {
 	// count how many time a path is used by instances in cluster
 	CountDir(host string, dir string) int
 
+	Component(name string) Component
```

Comment:
There's no need to add this to the interface. better keep interface small.
This can be:
```
func  Component(s Topology, name string) Component {
	for _, com := range s.ComponentsByStartOrder() {
		if com.Name() == name {
			return com
		}
	}
	return nil
}
```